library/std/src/sys/pal/sgx/abi/usercalls/alloc.rs

```diff
@@ -651,13 +651,29 @@ where
     /// the source. This can happen for dynamically-sized types such as slices.
     pub fn copy_to_enclave_vec(&self, dest: &mut Vec<T>) {
         if let Some(missing) = self.len().checked_sub(dest.capacity()) {
-            dest.reserve(missing)
+            dest.reserve(missing);
         }
         // SAFETY: We reserve enough space above.
         unsafe { dest.set_len(self.len()) };
         self.copy_to_enclave(&mut dest[..]);
     }
 
+    /// Copies the value from user memory and place it into `dest`.
+    ///
+    /// # Panics
+    /// This function panics if the destination doesn't have the same length as
+    /// the source.
+    pub fn copy_to_enclave_uninit(&self, dest: &mut [MaybeUninit<T>]) {
```

Comment:
Needing this “feels” wrong to me. You could do `impl<T: UserSafeSized> UserSafeSized for MaybeUninit<T>` and then just create a `[MaybeUninit<u8>]` in `read_buf`. However, of course the whole point of `UserSafe` is that you want to explicitly assume everything you copy out of userspace is initialized, so that would kind of defeat the point.

Reply:
If both of those options are unsatisfying to you, what would you recommend I do so this can impl `read_buf`?